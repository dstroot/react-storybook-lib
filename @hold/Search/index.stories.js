import React from "react";
import { storiesOf } from "@storybook/react";

// components
import Search from "../Search";

// Search section
storiesOf("Search", module)
  .addParameters({
    // default for all stories in this book
    info: {
      inline: true,
      header: false
    }
  })
  .add("show Search", () => <Search />);
