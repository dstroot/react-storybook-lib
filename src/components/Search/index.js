import React from "react";
import useMediaQuery from "../../hooks/useMediaQuery";
import image from "./media/financial_pro_2.jpg";
import Disclosure from "../Disclosure";

const textBig = {
  fontSize: `calc(24px + 1vw)`,
  marginBottom: "0",
  fontWeight: "300"
};

/**
Search is a component that is used to search for financial professionals near you.  
Note: This component uses the <Disclosure /> component.
 */
const Search = () => {
  // NOTE: do not use "@media" at start of media query
  const isTiny = useMediaQuery(`screen and (max-width: 200px)`);
  const isWide = useMediaQuery(`screen and (min-width: 768px)`);
  const isWider = useMediaQuery(`screen and (min-width: 900px)`);

  return (
    <div className="container mt-5">
      <div className="row align-items-center">
        {isWide && (
          <div className="col-md-6">
            <div className="thumbnail">
              <img
                className="img-fluid"
                src={image}
                alt="financial professional"
              />
            </div>
          </div>
        )}
        <div className="col-md-6">
          {!isTiny && (
            <h1 style={textBig}>CONFIDENCE IN YOUR FUTURE STARTS HERE</h1>
          )}
          <h3 className="">Find a Financial Professional</h3>
          {isWider && (
            <p className="lead">
              You want to reach your goals. A financial professional can help
              you get there. Search for a qualified professional near you.
            </p>
          )}
          <form>
            <div className="form-row align-items-center">
              <div className="col-auto">
                <input
                  aria-label="zip code"
                  type="text"
                  className="form-control mb-2"
                  id="zip"
                  placeholder="enter zip code"
                />
              </div>
              <div className="col-auto">
                <button type="submit" className="btn btn-primary mb-2">
                  Submit
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
      <Disclosure />
    </div>
  );
};

export default Search;
