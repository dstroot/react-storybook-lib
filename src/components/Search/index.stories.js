import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import Search from '../Search';

// Search section
storiesOf('Pac Life|Search', module)
  .addParameters({
    info: {
      inline: true,
      header: false,
    },
  })
  .add('show Search', () => <Search />);
