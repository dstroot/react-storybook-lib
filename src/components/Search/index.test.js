import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../Search';

describe('Search', () => {
  it('it should render', () => {
    const component = renderer.create(<Search />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
