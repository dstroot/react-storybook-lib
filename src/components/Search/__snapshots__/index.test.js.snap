// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search it should render 1`] = `
<div
  className="container"
>
  <div
    className="row align-items-center"
  >
    <div
      className="col-md-6"
    >
      <h1
        style={
          Object {
            "fontSize": "calc(24px + 1vw)",
            "fontWeight": "300",
            "marginBottom": "0",
          }
        }
      >
        CONFIDENCE IN YOUR FUTURE STARTS HERE
      </h1>
      <h3
        className=""
      >
        Find a Financial Professional
      </h3>
      <form>
        <div
          className="form-row align-items-center"
        >
          <div
            className="col-auto"
          >
            <input
              aria-label="zip code"
              className="form-control mb-2"
              id="zip"
              placeholder="enter zip code"
              type="text"
            />
          </div>
          <div
            className="col-auto"
          >
            <button
              className="btn btn-primary mb-2"
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
